scripts/jenkins/ardana/ansible/pre-deploy-checks.yml

```diff
@@ -0,0 +1,30 @@
+- name: Create a temporary directory to store pre-deployment verification state
+  become: yes
+  shell: mktemp -d /tmp/ardana-job-rpm-verification.XXXXXXXX
+  register: verification_mktemp_dir
+
+# This allows us to easily pass a value for the temporary directory
+# in to the test-post-deployment-checks.yml playbook when testing.
+- name: Store path to temporary directory
+  set_fact:
+    verification_temp_dir:
+      "{{ verification_mktemp_dir.stdout }}"
+
+- name: Set files to contain output of rpm -qf and -Va pre-deployment
+  set_fact:
+    pre_deploy_unowned_files:
+      "{{ verification_temp_dir }}/rpm-qf-pre.out"
+    pre_deploy_rpm_verification:
+      "{{ verification_temp_dir }}/rpm-verify-pre.out"
+
+- name: Take snapshot of files not owned by a package prior to deployment
+  become: yes
+  failed_when: false
+  shell: |
+    find /usr -print0 | xargs -0 rpm -qf | grep "is not owned" \
+      > "{{ pre_deploy_unowned_files }}"
+
+- name: Take snapshot of modified/deleted rpm-owned files prior to deployment
+  become: yes
+  shell: rpm -Va > "{{ pre_deploy_rpm_verification }}"
```

Comment:
I am also wondering that this task is only executed in the deployer, so we will miss all the wrongdoing in the control plane, compute nodes and storage.

Maybe we want to gather all this information (and the rest) in for host: * via some other playbook or role

Reply:
Very good point, thanks!  Urgh, that will require running the checks from the deployer though, since `ardana-ci-launcher` doesn't have access to the other nodes. Let's deal with that in a follow-up so as not to delay the immediate benefit this provides.